src/v/pandaproxy/client/consumer.cc

```diff
@@ -307,10 +321,114 @@ ss::future<kafka::offset_commit_response> consumer::offset_commit(
     return req_res(std::move(req_builder));
 }
 
-ss::future<shared_consumer_t>
-make_consumer(shared_broker_t coordinator, kafka::group_id group_id) {
+ss::future<kafka::fetch_response>
+consumer::consume(std::chrono::milliseconds timeout, int32_t max_bytes) {
+    using broker_reqs_t
```

Comment:
you aren't really leveraging the expense of flat_hash_map. i'd suggest moving to node_hash_map. 

the iteration on the map-reduce below invalidates any perf from the packed struct.